UDS Documentation Updates

jalling97 opened the pull request:
Quick PR to update some the documentation:
- Point people to use UDS to deploy LFAI
- Deprecate instructions for k8s deployment that doesn't use UDS
- Minor other doc fixes

The plan is to adjust this documentation (and add more) once #322 gets finalized, but this documentation is up to date at the moment.

justinthelaw commented on README.md:
```diff
@@ -32,34 +33,36 @@ Large Language Models (LLMs) are a powerful resource for AI-driven decision maki
+### UDS
 
-### Getting Started
+UDS is the preferrred method for a full local deployment of LeapfrogAI. Instructions can be found on the [LeapfrogAI Documentation Site](https://docs.leapfrog.ai/docs/).
```
Might be a good idea to mention that this is a Kubernetes deployment, even if our linked docs website mentions it later on.

jalling97 replied:
Updated!

YrrepNoj commented on README.md:
```diff
@@ -6,14 +6,15 @@
 - [Why Host Your Own LLM?](#why-host-your-own-llm)
+- [Getting Started](#getting-started)
+  - [UDS](#uds)
+  - [Tadpole](#tadpole)
```
We should remove the tadpole subsection, Tadpole is something we won't be supporting for a while.

YrrepNoj commented on README.md:
```diff
@@ -32,34 +33,32 @@ Large Language Models (LLMs) are a powerful resource for AI-driven decision maki
 
-> GitHub Repos:
-> | Repo | AMD64 Support | ARM64 Support | Cuda Support | Docker Ready | K8s Ready | Zarf Ready |
+> Available Backends:
```
We should update these to link to their place within the monorepo

YrrepNoj commented on website/content/en/docs/local deploy guide/quick_start.md:
```diff
@@ -0,0 +1,86 @@
+
+## Prerequisites
+
+- [K3D](https://k3d.io/)
```
k3d implies Docker but their instructions don't have you install docker. we can list `Docker` here to be explicit.

I wouldn't hate it if we didn't though.

gphorvath replied:
yeah we probably should list it

YrrepNoj commented on website/content/en/docs/local deploy guide/quick_start.md:
```diff
@@ -0,0 +1,86 @@
+
+## Disclaimers
+
+It is recommended to run UDS as root.
```
Hmm? I don't think this is true?

YrrepNoj replied:
As far as I know, the only time root is recommended (required) is when you are installing a `k3s` cluster when running the `zarf init` command.

YrrepNoj commented on website/content/en/docs/local deploy guide/quick_start.md:
```diff
@@ -0,0 +1,86 @@
+
+GPU workloads **_WILL NOT_** run if GPU resources are unavailable to the pod(s). You must provide sufficient NVIDIA GPU scheduling or else the pod(s) will go into a crash loop.
+
+`whisper` can run without without GPU scheduling - just turn off the `GPU_ENABLED` boolean.
```
```suggestion
`whisper` can run without GPU scheduling - just set the `GPU_LIMIT` value to `0`.
```

YrrepNoj commented on website/content/en/docs/local deploy guide/quick_start.md:
````diff
@@ -0,0 +1,86 @@
+``` bash
+cd bundles/cpu/
+uds create .
+uds deploy k3d-core-istio-dev:0.14.1
````
Maybe add a comment to check if a newer version exists.

jalling97 commented:
As a last minute edit, I removed references to Tadpole from the deploy instructions in the docs and moved the Tadpole specific instructions to a draft so that it won't render on the site. This way if we do support Tadpole again the instructions will still be there but not actually visible. Let me know if that's a sufficient approach for the time being.

YrrepNoj commented on website/content/en/docs/local deploy guide/quick_start.md:
````diff
@@ -0,0 +1,85 @@
+``` bash
+cd bundles/cpu/
+uds create .
+uds deploy k3d-core-istio-dev:0.14.1      # be sure to check if a newer version exists
````
This is a public image so that shouldn't be a hard requirement. We could always suggest it as a best practice still.

justinthelaw replied:
Hm, I actually tried it out and I wasn't able to pull it without logging in with a valid PAT.

jalling97 replied:
I personally haven't run into that issue, but I can add some text. I'm assuming the PAT needs at least `read:packages` access?

justinthelaw replied:
Hm, I realized my PAT had expired on Friday and I had to make a new PAT and re-login via CLI in order to pull packages from GHCR. It is possible an expired PAT will cause a 403 error purely because it tries to authorize if a PAT is available in cred store. I think this is fine without the PAT!
